### PR TITLE
bjorck2018understanding

### PR DESCRIPTION
**Pull request by timchen0618**
very shallow intro (need to make a lot more edits)

### CODE REVIEW

File: _summaries/bjorck2018understanding_1.md [hunk 1]
```diff
@@ -1,19 +1,63 @@
...
-* What interesting variants are explored?
+## Background
+Normalizing the input data to zero-mean and constant standard deviation has long been recognized as good practice in training neural networks. 
+Batch normalization (BN) extend this idea to the input of each layer and achieve immense success in various areas. 
+It enables faster training, larger learning rates, and higher testing accuracy (and better generalizability). 
+However, the reason for the improvement is to be debated. 
+(e.g. the paper that propose BN said it was solving "internal covariate shift", but some suggested otherwise)
+
+## Core idea 
+The paper pointed out that BN is beneficial to training neural networks mainly because it allow for larger learning rates.
```
Comment (specfazhou) [inline _summaries/bjorck2018understanding_1.md:19]: I guess it will be good to mention that since network with BN and network without BN, if they have same learning rate, they can achieve same level of accuracy, which indicates that the success of BN is because it enables the use of higher learning rate.

File: _summaries/bjorck2018understanding_1.md [hunk 2]
```diff
@@ -1,19 +1,63 @@
...
+## Background
+Normalizing the input data to zero-mean and constant standard deviation has long been recognized as good practice in training neural networks. 
+Batch normalization (BN) extend this idea to the input of each layer and achieve immense success in various areas. 
+It enables faster training, larger learning rates, and higher testing accuracy (and better generalizability). 
+However, the reason for the improvement is to be debated. 
+(e.g. the paper that propose BN said it was solving "internal covariate shift", but some suggested otherwise)
+
+## Core idea 
+The paper pointed out that BN is beneficial to training neural networks mainly because it allow for larger learning rates. 
+It also showed empirically that larger learning rates in networks without BN could lead to strong gradient updates, exploding activations and diverging loss . The same cannot be observed for neural networks with batch normalization, since the activation will be contained within a certain range, hence the stable loss and gradient.
```
Comment (specfazhou) [inline _summaries/bjorck2018understanding_1.md:20]: I am a little confused here, in my understanding, I think large learning makes networks without BN diverge, but since networks with BN still converges with large learning rate, and by the fact that larger learning rate cause more noise when using SGD method, which makes prevents network from getting trapped in sharp minima, and get better generalization. Sorry if I understand this wrong. I guess maybe it will be nice if you can write more details here.

File: _summaries/bjorck2018understanding_1.md [hunk 3]
```diff
@@ -1,19 +1,63 @@
...
+
+## Core idea 
+The paper pointed out that BN is beneficial to training neural networks mainly because it allow for larger learning rates. 
+It also showed empirically that larger learning rates in networks without BN could lead to strong gradient updates, exploding activations and diverging loss . The same cannot be observed for neural networks with batch normalization, since the activation will be contained within a certain range, hence the stable loss and gradient. 
+
+## Experimental Results
+
+### Why is BN beneficial?
+As in the above figure, training with BN but with a small learning rate (lr=0.0001) does not improve testing accuracy. 
+Additionally, testing accuracy is higher when they chose large learning rate (0.1), displaying better generalization.
```
Comment (specfazhou) [inline _summaries/bjorck2018understanding_1.md:26]: it seems this sentence doesn't explain why BN beneficial.

File: _summaries/bjorck2018understanding_1.md [hunk 4]
```diff
@@ -1,19 +1,63 @@
...
+
+### Divergence of NN Without BN
+Here they define 
+- relative loss (step-size) = new_loss/old_loss. 
+- divergence: when relative loss > $$10^3$$.
+At first few updates, network without BN exhibit great relative loss.
+![Step Size](bjorck2018understandin_1b.png)
+Activation of upper layers are extremely large, several orders of magnitude larger than lower ones. (Notice the scale)
+The authors claim that this implies divergence is caused by exploding activations.
+And BN fixes this problem by normalizing the input, preventing the large activations from propagating.
```
Comment (specfazhou) [inline _summaries/bjorck2018understanding_1.md:38]: I guess it is also worth to mention that means and variances of channels are quite constant on the log-scale as a function of depth of layers. and can also add a comment saying that normalizing the last layer might be the most important contribution of BN

**Review by specfazhou — COMMENTED** [review]
it is a good review overall, just need some small modifications

File: _summaries/bjorck2018understanding_1.md (hunk 2 above)
Comment (timchen0618) [reply _summaries/bjorck2018understanding_1.md:20]: I think what I am talking about here is why networks with BN still converges on large learning rates. They first showed empirically that BN is beneficial because of the large learning rate they use, and later showed why such large rates can be used. The generalizability is more of the theoretical reason why BN is beneficial, but yeah maybe I should include that as well. (Was gonna put it in later sections)

File: _summaries/bjorck2018understanding_1.md (hunk 4 above)
Comment (timchen0618) [reply _summaries/bjorck2018understanding_1.md:38]: The first sentence I think is a good idea! I'll include that.

Comment (timchen0618) [reply _summaries/bjorck2018understanding_1.md:38]: For the second sentence, I think it also mentioned in the paper that "Figure 6 also shed light onto why the gradients of networks without BN tend to be so large in the final layers: the rows of the heatmap are highly correlated". In my understanding, BN solves it by distributing the gradients across classes, leading to the observation that final layers are normalized. Does it make sense?

File: _summaries/bjorck2018understanding_1.md (hunk 1 above)
Comment (timchen0618) [reply _summaries/bjorck2018understanding_1.md:19]: yeah! That would be more clear, thank you!

**Comment by timchen0618** [comment]
I have modified my summary, can you help me see if there are any points that are not clear? (Especially ones you commented on ) Also, I am afraid that it might be too long, do you think it will be a problem? Should I just get rid of the "Gradient With or Without BN" subsection? (Guess it's not that important for the point the author is trying to make)

File: _summaries/bjorck2018understanding_1.md [hunk 5]
```diff
@@ -1,19 +1,92 @@
...
+
+![Activation Heat Map](bjorck2018understandin_1c.png)
+
+As seen in the figure below, when BN is applied, the activations are indeed contained within a roughly constant value across layers. While the mean and standard deviation of channels grow exponentially with depth.
+
+![Channel Mean and Variance](bjorck2018understandin_1d.png)
+
+
+### Gradients With and Without BN
```
Comment (timchen0618) [inline _summaries/bjorck2018understanding_1.md:76]: Do you think this subsection is needed? I'm not sure if the whole summary will be too long.

Comment (specfazhou) [reply _summaries/bjorck2018understanding_1.md:79]: It is a pretty good summary to me right now, great!

Comment (timchen0618) [reply _summaries/bjorck2018understanding_1.md:79]: So you don't think it'll be too long I guess?

Comment (specfazhou) [reply _summaries/bjorck2018understanding_1.md:79]: I think it is not too long, should be fine
